src/erlmld_wrk_statem.erl

```diff
@@ -460,14 +460,24 @@ handle_event(?INTERNAL,
             throw({stop, {error, {unprocessed_request_data, Buf}}})
     end,
     ok = gen_tcp:send(Socket, [IoData, "\n"]),
-    {next_state, {?PEER_READ, NextReadKind}, activate(Data)};
+    case NextReadKind of
+        Kind when Kind == ?SHUTDOWN; Kind == ?SHUTDOWN_CHECKPOINT ->
```

Comment:
I think it only needs to be done when the `NextReadKind` is `?SHUTDOWN`.
The `?SHUTDOWN_CHECKPOINT` state happens right before the `SHUTDOWN` state.